### DISCUSSION TITLE
Changelog for replicated docker images of Github project

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#10338] [Request Help] [answered]
The lab where our Renovate instances run don't have access to the public internet. For that reason, we do the[ "docker pull, tag and push" dance](https://stackoverflow.com/a/28349540) to replicate a selection of public docker images.

For the changelog in the MR, `getChangeLogJSON` in `lib/workers/pr/changelog/index.ts` invokes a GitHub API, but it seemingly filters out all the images that aren't preset in the (in this case private) docker repo.

In the case of Renovate itself, there are so many/frequent change, and the layers are so large, that soon it becomes impractical to keep the Renovate pipeline up to date with Renovate.

Why is the GitHub projects changelog filtered by what's present in Docker?

**Comment by rarkins** [comment]
I suggest you look into modifying the PR body columns or values and add a link like https://octoclairvoyant.vercel.app/?repo=renovatebot%2Frenovate&from=23.95.0&to=24.22.3

**Comment by astellingwerf** [comment, marked as answer]
For anyone who's interested, I got this to work by creating a custom packageRule, that inserts a link in the place of the Release Notes:
![image](https://user-images.githubusercontent.com/1637358/121090117-5fe33d00-c7e8-11eb-89d1-58dfee9e82a8.png)

This is the custom packageRule:
```typescript
    // Dependencies that we mirror from Docker Hub into Artifactory
    {
      matchManagers: ['gitlabci'],
      matchPackagePatterns: ['/renovate/renovate$'],

      // Delete the generated changelogs, as those are very incomplete
      prBodyTemplate: '{{{header}}}{{{table}}}{{{notes}}}{{{configDescription}}}{{{controls}}}{{{footer}}}',
      // Instead, add a link in the Notes section
      prBodyNotes: [
          '---\n\n### Release Notes\n\nSee [:octopus: Octoclairvoyant :crystal_ball:](https://octoclairvoyant.vercel.app/?repo=renovatebot%2Frenovate&from={{{currentVersion}}}&to={{{newVersion}}}) for an overview of changes between versions {{{displayFrom}}} and {{{displayTo}}}.',
      ],
    }
```
